File: articles/positive-psychology-interventions/c/ppi_mechanism_index.c
#include <stdio.h>

typedef struct {
    double wellbeing_score;
    double depressive_symptoms;
    double gratitude_score;
    double strengths_use;
    double hope_score;
    double meaning_score;
    double social_support;
    double adherence_rate;
    double intervention_fit;
    double stress_load;
    double acceptability;
    double context_fit;
} PPIIndicators;

double mechanism_index(PPIIndicators x) {
    return (x.gratitude_score + x.strengths_use + x.hope_score + x.meaning_score + x.social_support) / 5.0;
}

double practice_quality_proxy(PPIIndicators x) {
    return (x.adherence_rate + x.intervention_fit + x.acceptability + x.context_fit) / 4.0;
}

double net_wellbeing_index(PPIIndicators x) {
    return x.wellbeing_score + x.gratitude_score + x.strengths_use + x.hope_score + x.meaning_score + x.social_support + x.intervention_fit - x.stress_load - x.depressive_symptoms;
}

int main(void) {
    PPIIndicators example = {7.2, 3.8, 7.3, 6.5, 6.6, 6.7, 7.0, 0.88, 7.4, 3.4, 7.5, 7.4};
    printf("Mechanism index: %.3f\n", mechanism_index(example));
    printf("Practice quality proxy: %.3f\n", practice_quality_proxy(example));
    printf("Net wellbeing index: %.3f\n", net_wellbeing_index(example));
    return 0;
}

File: articles/positive-psychology-interventions/fortran/ppi_mechanism_index.f90
program ppi_mechanism_index
  implicit none
  real :: wellbeing_score, depressive_symptoms, gratitude_score, strengths_use
  real :: hope_score, meaning_score, social_support, adherence_rate
  real :: intervention_fit, stress_load, acceptability, context_fit
  real :: mechanism, practice_quality, net_wellbeing

  wellbeing_score = 7.2
  depressive_symptoms = 3.8
  gratitude_score = 7.3
  strengths_use = 6.5
  hope_score = 6.6
  meaning_score = 6.7
  social_support = 7.0
  adherence_rate = 0.88
  intervention_fit = 7.4
  stress_load = 3.4
  acceptability = 7.5
  context_fit = 7.4

  mechanism = (gratitude_score + strengths_use + hope_score + meaning_score + social_support) / 5.0
  practice_quality = (adherence_rate + intervention_fit + acceptability + context_fit) / 4.0
  net_wellbeing = wellbeing_score + gratitude_score + strengths_use + hope_score + meaning_score + social_support + intervention_fit - stress_load - depressive_symptoms

  print *, "Mechanism index:", mechanism
  print *, "Practice quality proxy:", practice_quality
  print *, "Net wellbeing index:", net_wellbeing
end program ppi_mechanism_index
